packages/contracts/src/JwtAuth.sol

```diff
@@ -0,0 +1,142 @@
+    IVerifier internal verifier;
+    /// A mapping of the hash of the authorized message associated with its `jwtNullifier`.
+    mapping(bytes32 => bool) public usedNullifiers;
+
+    event JwtRegistryUpdated(address indexed jwtRegistry);
+    event VerifierUpdated(address indexed verifier);
+    event JwtAuthed(
+        bytes32 indexed jwtNullifier,
+        bool isCodeExist,
+        string indexed domainName,
+        string indexed azp 
+    );
+
+    constructor() {}
+
+    /// @notice Initialize the contract with an initial owner and an account salt.
+    /// @param _initialOwner The address of the initial owner.
+    /// @param _accountSalt The account salt to derive CREATE2 address of this contract.
+    function initialize(
+        address _initialOwner,
+        bytes32 _accountSalt
+    ) public initializer {
+        __Ownable_init(_initialOwner);
+        accountSalt = _accountSalt;
+    }
+
+    /// @notice Initializes the address of the JWT registry contract.
+    /// @param _jwtRegistryAddr The address of the JWT registry contract.
+    function initJwtRegistry(address _jwtRegistryAddr) public onlyOwner {
+        require(
+            _jwtRegistryAddr != address(0),
+            "invalid jwt registry address"
+        );
+        require(
+            address(jwtRegistry) == address(0),
+            "jwt registry already initialized"
+        );
+        jwtRegistry = JwtRegistry(_jwtRegistryAddr);
+        emit JwtRegistryUpdated(_jwtRegistryAddr);
+    }
+
+    /// @notice Initializes the address of the verifier contract.
+    /// @param _verifierAddr The address of the verifier contract.
+    function initVerifier(address _verifierAddr) public onlyOwner {
+        require(_verifierAddr != address(0), "invalid verifier address");
+        require(
+            address(verifier) == address(0),
+            "verifier already initialized"
+        );
+        verifier = IVerifier(_verifierAddr);
+        emit VerifierUpdated(_verifierAddr);
+    }
+
+    /// @notice Updates the address of the JWT registry contract.
+    /// @param _jwtRegistryAddr The new address of the JWT registry contract.
+    function updateJwtRegistry(address _jwtRegistryAddr) public onlyOwner {
+        require(
+            _jwtRegistryAddr != address(0),
+            "invalid dkim registry address"
+        );
+        jwtRegistry = JwtRegistry(_jwtRegistryAddr);
+        emit JwtRegistryUpdated(_jwtRegistryAddr);
+    }
+
+    /// @notice Updates the address of the verifier contract.
+    /// @param _verifierAddr The new address of the verifier contract.
+    function updateVerifier(address _verifierAddr) public onlyOwner {
+        require(_verifierAddr != address(0), "invalid verifier address");
+        verifier = IVerifier(_verifierAddr);
+        emit VerifierUpdated(_verifierAddr);
+    }
+
+    function authJwt(JwtProof memory proof) public onlyOwner {
+
+        // require(
+        //     accountSalt == proof.accountSalt,
```

Comment:
@shreyas-londhe 
For jwt-tx-builder, I want to make sure that this process is not necessary because I do not think that the specification to create a JwtAuth contract for each different accountSalt. If you want to create a JwtAuth for each domainName or azp, please let me know